lib/psabpf_table.c

```diff
@@ -117,6 +200,24 @@ int psabpf_table_entry_ctx_tblname(psabpf_context_t *psabpf_ctx, psabpf_table_en
         fprintf(stderr, "warning: cache for table %s not found: %s\n", name, strerror(ret));
     }
 
+    if (ctx->btf_metadata.btf == NULL || ctx->table.btf_type_id == 0) {
+        fprintf(stderr, "unable to handle direct objects; resetting them if exist\n");
+    } else {
+        ret = execute_direct_objects_command(ctx, DIRECT_OBJECT_COUNT);
+        if (ret == NO_ERROR) {
+            if (ctx->n_direct_counters > 0) {
+                ctx->direct_counters_ctx = malloc(ctx->n_direct_counters * sizeof(psabpf_direct_counter_context_t));
+                for (unsigned i = 0; i < ctx->n_direct_counters; i++)
+                    psabpf_direct_counter_ctx_init(&ctx->direct_counters_ctx[i]);
```

Comment:
`psabpf_direct_counter_ctx_init` and `DIRECT_OBJECT_INIT_CTX` command confuse me

Reply:
Moved to `init_direct_objects`